test/e2e/persistent_volumes.go

```diff
@@ -114,14 +114,36 @@ func deletePVCandValidatePV(c *client.Client, ns string, pvc *api.PersistentVolu
 	return pv, pvc, nil
 }
 
+// Wraps deletePVCandValidatePV() to delete and validate 1:1 PV-PVC pairs
+// Takes the list of pvs and list of pvcs
+// Returns updated pvList, pvcList, and error
+func deleteMultiplePVCandValidatePV(c *client.Client, ns string, pvcList []*api.PersistentVolumeClaim, pvList []*api.PersistentVolume) ([]*api.PersistentVolume, []*api.PersistentVolumeClaim, error) {
+
+	var err error
+
+	for pvi, pv := range pvList {
+		for pvci, pvc := range pvcList {
+			if pvcList[pvci] != nil && len(pvc.Name) > 0 {
```

Comment:
1) When I've used `if pvc != nil.....` on line 126, the call returns after 1 iteration with remaining PVCs.  I haven't uncovered the reason for this yet but it performs as expected if I directly access the slice by index.
2) The nil check is there to skip deleted PVCs.  As the loop iterates, the call returns a nil value (as result of `deletePVCandValidate()` -> `pvc, err = c.PersistentVolumeClaims(ns).Get(pvc.Name)`.  Without it, an attempt to read pvc.UID causes a nil pointer error and stack trace.